### ISSUE TITLE
Half/mixed-precision is not faster

### ISSUE THREAD

**Issue by as1392** [facebookresearch/maskrcnn-benchmark#1062]
## ❓ Questions and Help
As written in https://github.com/facebookresearch/maskrcnn-benchmark/issues/807#issuecomment-500112612 , DTYPE "float16" does not make training faster( and takes the same amount of memory) when tested with e2e_mask_rcnn_R_50_FPN_1x.yaml. Is this normal or am I doing something wrong? I also want to know if I can make inference faster by FP16(I tried model.half() in test_net, but doesn't work).  
My environment is RTX Titan, cudnn 7.6, cuda10.0, driver version Driver Version: 418.67, pytorch-1.2.0, ubuntu 18.04.

**Comment by Jacobew** [comment]
Any update here? I've observed the same issue using fp16. The memory footprint and training speed seems to have no difference. Are there any specific requirements for PyTorch and cudnn versions?  @fmassa @slayton58

**Comment by cestxiaoxi** [comment]
Yes, I am facing this phenomenon too (with the latest CUDA and CUDNN versions). It would be nice that someone give us some pointers on where we should be looking.

**Comment by Kobemaomingyuan** [comment]
> ❓ Questions and Help
> As written in #807 (comment) , DTYPE "float16" does not make training faster( and takes the same amount of memory) when tested with e2e_mask_rcnn_R_50_FPN_1x.yaml. Is this normal or am I doing something wrong? I also want to know if I can make inference faster by FP16(I tried model.half() in test_net, but doesn't work).
> My environment is RTX Titan, cudnn 7.6, cuda10.0, driver version Driver Version: 418.67, pytorch-1.2.0, ubuntu 18.04.

Have you met the error when using "float16":
 File "/home/mmy/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/nn/modules/module.py", line 493, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/mmy/tiny-person-competition/cur_tinybench/TinyBenchmark-master/tiny_benchmark/maskrcnn_benchmark/layers/dcn/deform_conv_module.py", line 59, in forward
    self.deformable_groups)
  File "/home/mmy/tiny-person-competition/cur_tinybench/TinyBenchmark-master/tiny_benchmark/maskrcnn_benchmark/layers/dcn/deform_conv_func.py", line 66, in forward
    cur_im2col_step
RuntimeError: expected scalar type Float but found Half (data<float>)
